roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -95,29 +95,30 @@ impl JDsMempool {
     }
 
     pub async fn update_mempool(self_: Arc<Mutex<Self>>) -> Result<(), JdsMempoolError> {
-        let mut mempool_ordered: HashMap<Txid, Option<Transaction>> = HashMap::new();
+        let mut new_jds_mempool: HashMap<Txid, Option<Transaction>> =
+            self_.safe_lock(|x| x.mempool.clone())?;
+
+        // the fat transactions in the jds-mempool are those declared by some downstream and we
+        // don't want to remove them, but we can get rid of transactions that are none
+        new_jds_mempool.retain(|_, val| val.is_some());
```

Comment:
What if the txs declared in the job are still not inserted in the mempool, but this function is executed? Are they gonna by deleted or not?

Reply:
If a transaction has not yet been inserted and its value is None, this line would remove it from the mempool

Reply:
would you suggest we refrain from removing transactions that are none? I am doing this so that there is not an overflow in the jds mempool of old transactions

Reply:
Aren't you managing the old transactions removal with the clear_old_mempool_transactions? Is it not enough in your opinion?

Reply:
The clear_old_mempool_transactions handles the removal of old transactions, ensuring the mempool is kept clean and up-to-date. While this retain function would  ensures that only transactions that have been fully fetched and are valid remain in the mempool. Isn't this what we want to achieve? Except maybe there is some edge case I am missing or maybe I might need clarity because the later part of the update_mempool function updates the mempool and ensures transaction IDs that are not in the JDS mempool are added along with their transaction data

Reply:
I feel this should be enough

Reply:
IMO the two functions have two purposes.
`clear_old_mempool_transactions` is called when a new job arrived and removes the transactions of the old job
`update_mempool` keep in sync the jds light mempool with the node mempool. You want to keep the transactions that are declared by some downstream, because there is the possibility that the node is not aware of some of them. BTW in the future I'll write an rpc  method to push txs to the node when they arrive to the jds, this improves a lot

Reply:
This means there would be no need for the filtration I am doing on the JDS mempool? @lorbax